### PR TITLE
Add field usage support for vectors

### PR DESCRIPTION
**Pull request by ywelsch**
Add field usage tracking support for the new vectors functionality.

### CODE REVIEW

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/50_dense_vector_field_usage.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+setup:
```
Comment (cbuescher) [inline x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/50_dense_vector_field_usage.yml:1]: This test might need a skip section for <8.1 for mixed cluster tests.

**Review by cbuescher — APPROVED** [review]
LGTM, left one note regarding possible problem with the rest test for mixed cluster tests

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/50_dense_vector_field_usage.yml (hunk 1 above)
Comment (cbuescher) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/50_dense_vector_field_usage.yml:1]: hmm, don't really understand why this passes test. Maybe we don't run these in x-pack on mixed clusters?

Comment (ywelsch) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/50_dense_vector_field_usage.yml:1]: yes, this validates my understanding (I deliberately did not add the version constraint to check if anything breaks on CI). I find the findings very worrying though.

Comment (cbuescher) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/50_dense_vector_field_usage.yml:1]: Just see this passed, so nevermind. I don't really know why this works in mixed cluster tests wasn't aware that yml in x-pack isn't run against

Comment (cbuescher) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/50_dense_vector_field_usage.yml:1]: Good to see that we both expected this to fail in some way. Maybe worth some parallel investigation, I'd be interested in the results.

File: docs/reference/indices/field-usage-stats.asciidoc [hunk 2]
```diff
@@ -128,7 +128,8 @@ The API returns the following response:
                         "doc_values" : 1,
                         "points" : 0,
                         "norms" : 1,
-                        "term_vectors" : 0
+                        "term_vectors" : 0,
+                        "vectors" : 0
```
Comment (jtibshirani) [inline docs/reference/indices/field-usage-stats.asciidoc:132]: Maybe we could call this "knn_vectors". This matches the naming in Lucene (`KnnVectorField`, `KnnVectorsFormat`). In other places, we've received feedback that "vector" is too ambiguous in the context of Elasticsearch/ Lucene.

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/50_dense_vector_field_usage.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                index: true
+                similarity: l2_norm
+
+  - do:
+      index:
+        index: futest
+        body:
+          name: cow.jpg
+          vector: [ 230.0, 300.33, -34.8988, 15.555, -200.0 ]
+          comments:
```
Comment (jtibshirani) [inline x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/50_dense_vector_field_usage.yml:29]: The `comments` part of these documents can be removed. I accidentally forgot to clean this up when I simplified the mapping in https://github.com/elastic/elasticsearch/pull/80516.

Comment (ywelsch) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/50_dense_vector_field_usage.yml]: 6e42856 👍

File: docs/reference/indices/field-usage-stats.asciidoc (hunk 2 above)
Comment (ywelsch) [reply docs/reference/indices/field-usage-stats.asciidoc]: 6e42856 👍

**Review by jtibshirani — APPROVED** [review]

File: x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/50_dense_vector_field_usage.yml (hunk 1 above)
Comment (mark-vieira) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/50_dense_vector_field_usage.yml:1]: It seems we don't have "mixed cluster" tests for x-pack like we do for the core apis. So we simply aren't running this test under that scenario at all.

https://github.com/elastic/elasticsearch/issues/31096

Comment (ywelsch) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/50_dense_vector_field_usage.yml:1]: Wow, that's scary (and lol for linking to my own issue, guess my memory is failing me).

Perhaps that one needs to be reprioritized!

Comment (mark-vieira) [reply x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/50_dense_vector_field_usage.yml:1]: The issue seems to be tagged as core/infra but I'm not sure if that's strictly correct. There's no real distinct ownership of these large cross-functional test suites. It's really just a matter of adding a qa project for this under x-pack similar to the existing core one.

**Pull request merged**